utils/prometheus/processor.py

```diff
@@ -0,0 +1,95 @@
+# (C) Datadog, Inc. 2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import requests
+from utils.prometheus import parse_metric_family
+
+
+# message.type is the index in this array
+# see: https://github.com/prometheus/client_model/blob/master/ruby/lib/prometheus/client/model/metrics.pb.rb
+METRIC_TYPES = ['counter', 'gauge', 'summary', 'untyped', 'histogram']
+
+
+class PrometheusProcessor:
+    def __init__(self, conf):
+        # metrics_mapper is a dictionnary where the keys are the metrics to capture and the values are the corresponding metrics names to have in datadog.
+        self.metrics_mapper = {}
+        # send_histograms_buckets is used to specify if yes or no you want to send the buckets as tagged values when dealing with histograms.
+        self.send_histograms_buckets = True
+        # endpoint is the metrics endpoint to use to poll metrics from Prometheus
+        self.endpoint = 'http://localhost:10055/metrics'
```

Comment:
Sure, I was thinking about overwriting it like that in the check method:
```        endpoint = instance.get('prometheus_endpoint')
        if endpoint is None:
            raise CheckException("Unable to find prometheus_endpoint in config file.")
        self.processor.endpoint = endpoint```
But having it as a parameter should do it. :)